### PR TITLE
typings(MessageEmebd): fix typings for addFields

### PR DESCRIPTION
**Pull request by almostSouji**
**Please describe the changes this PR makes and why it should be merged:**

Fix typings for `addFields `and `spliceFields `as well as the underlying `normalizeFields `call by introducing a second typing, **EmbedFieldData** alongside EmbedField to enable passing of **StringResolvable** to the builder methods while keeping the certain string for the accessor through the field structure e.g. `embed.fields[0].name` as well as a certain boolean for the inline accessor - as inline is defaulted to `false` in the normalizeField call

⚠ I have tested this with all possible inputs and accessors I could think of, however please let somebody proficient in typings review this before merging

**Status**
- [x] Code changes have been tested against the Discord API, or there are no code changes
- [x] I know how to update typings and have done so, or typings don't need updating

**Semantic versioning classification:**  
- [ ] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: typings/index.d.ts [hunk 1]
```diff
@@ -2184,7 +2184,13 @@ declare module 'discord.js' {
...
 		name: string;
 		value: string;
-		inline?: boolean;
+		inline: boolean;
+	}
+
+	interface EmbedFieldData {
+		name: StringResolvable;
+		value: StringResolvable;
+		inline?: boolean
```
Comment (NotSugden) [inline typings/index.d.ts]:
```suggestion
		inline?: boolean;
```
missing semi colon

**Comment by NotSugden** [comment]
shouldn't there also be some changes for the jsdoc for EmbedField?

File: src/structures/MessageEmbed.js [hunk 2]
```diff
@@ -359,9 +359,16 @@ class MessageEmbed {
     return { name, value, inline };
   }
 
+  /**
+    * @typedef {Object} EmbedFieldData
+    * @property {StringResolvable} name The name of this field
+    * @property {StringResolvable} value The value of this field
+    * @property {?boolean} inline If this field will be displayed inline
```
Comment (SpaceEEC) [inline src/structures/MessageEmbed.js]: It's optional, not nullable :eyes:
```suggestion
    * @property {boolean} [inline] If this field will be displayed inline
```

**Comment by mrjackyliang** [comment]
Not sure if I should be creating a new issue, but why is `EmbedFieldData[] | EmbedFieldData[][]` not assignable to type `any`?

Screen Shot 2021-04-03 at 2 53 53 PM

**Pull request merged**